src/libos/crates/jindisk/src/util/cryption.rs

```diff
@@ -25,28 +25,32 @@ pub type Iv = [u8; AUTH_ENC_IV_SIZE];
 
 /// Encryption/Decryption.
 pub trait Cryption {
-    /// Encrypt a block.
-    fn encrypt_block(plaintext: &[u8], key: &Key) -> CipherBlock;
+    /// Encrypt a block. (AEAD)
+    fn encrypt_block_aead(plaintext: &[u8], key: &Key) -> CipherBlock;
 
-    /// Decrypt a block.
-    fn decrypt_block(ciphertext: &[u8], key: &Key, meta: &CipherMeta) -> Result<[u8; BLOCK_SIZE]>;
+    /// Decrypt a block. (AEAD)
+    fn decrypt_block_aead(
+        ciphertext: &[u8],
+        key: &Key,
+        meta: &CipherMeta,
+    ) -> Result<[u8; BLOCK_SIZE]>;
 
-    /// Encrypt content of any length.
-    fn encrypt_arbitrary(plaintext: &[u8], ciphertext: &mut [u8], key: &Key) -> CipherMeta;
+    /// Encrypt content of any length. (AEAD)
+    fn encrypt_arbitrary_aead(plaintext: &[u8], ciphertext: &mut [u8], key: &Key) -> CipherMeta;
 
-    /// Decrypt content of any length.
-    fn decrypt_arbitrary(
+    /// Decrypt content of any length. (AEAD)
+    fn decrypt_arbitrary_aead(
         ciphertext: &[u8],
         plaintext: &mut [u8],
         key: &Key,
         meta: &CipherMeta,
     ) -> Result<()>;
 
-    /// Encrypt a block using symmetric key cipher(AES128 CTR mode)
-    fn symm_encrypt_block(data: &[u8], key: &Key) -> Result<[u8; BLOCK_SIZE]>;
+    /// Encrypt a block. (Non-AEAD)
+    fn encrypt_block(data: &[u8], key: &Key) -> Result<[u8; BLOCK_SIZE]>;
 
-    /// Decrypt a block using symmetric key cipher(AES128 CTR mode)
-    fn symm_decrypt_block(data: &[u8], key: &Key) -> Result<[u8; BLOCK_SIZE]>;
+    /// Decrypt a block. (Non-AEAD)
+    fn decrypt_block(data: &[u8], key: &Key) -> Result<[u8; BLOCK_SIZE]>;
```

Comment:
The current crypto API encourages copying by returning `[u8; BLOCK_SIZE]`. Usually, the user of such API can prepare the target enc/dec buffer before calling the API. So by accepting `&mut [u8]` as input, the API can save the user from one extra copy of a block of memory.

Reply:
The start point to design this interface is to ensure the invariant of block-granularity cryption. There is a non-copy version called `encrypt_arbitrary(plaintext, ciphertext)` can serve the purpose.

And no obvious performance degradation according to some microbenchmarks by this PR.